src/sqlfluff/rules/L027.py

```diff
@@ -45,13 +47,33 @@ def _lint_references_and_aliases(
         using_cols,
         parent_select,
     ):
+        # Config type hints
+        self.ignore_words_regex: str
+
         # Do we have more than one? If so, all references should be qualified.
         if len(table_aliases) <= 1:
             return None
+
+        # Get the ignore_words_list configuration.
+        try:
+            ignore_words_list = self.ignore_words_list
+        except AttributeError:
+            # First-time only, read the settings from configuration. This is
+            # very slow.
+            ignore_words_list = self._init_ignore_words_list()
```

Comment:
@barrywhart question for you.

Should this use `memory` in any returns, like L010 to save the cached config?

Never really understood that, and I notice that L059 doesn't use this (but maybe should?)

I notice

Reply:
I prefer this technique for cached configuration, and `memory` for cached rule state (i.e. what happened earlier in the SQL file).

This caching technique is appropriate when executing complex code to get the rule configuration, rather than a value provided directly by the config as is typical. Here's the L010 code. Note how it involves loops, etc. Executing this code every time `_eval()` is called is **very** expensive.

```
        cap_policy_name = next(
            k for k in self.config_keywords if k.endswith("capitalisation_policy")
        )
        self.cap_policy = getattr(self, cap_policy_name)
        self.cap_policy_opts = [
            opt
            for opt in get_config_info()[cap_policy_name]["validation"]
            if opt != "consistent"
        ]
        # Use str() as L040 uses bools which might otherwise be read as bool
        ignore_words_config = str(getattr(self, "ignore_words"))
        if ignore_words_config and ignore_words_config != "None":
            self.ignore_words_list = self.split_comma_separated_string(
                ignore_words_config.lower()
            )
```